python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1226 @@
+        donutData["fx"] = self._getMetadataArray(donutStamps.metadata, "FX", numDonuts, 0.0)
+        donutData["fy"] = self._getMetadataArray(donutStamps.metadata, "FY", numDonuts, 0.0)
+        donutData["snr"] = self._getMetadataArray(donutStamps.metadata, "SN", numDonuts, np.nan)
+
+        # Add detector information as a column (required by downstream
+        # aggregation)
+        # Ensure detector name is consistent with expected format
+        donutData["detector"] = [detectorName] * numDonuts
+        self.log.info(
+            "Using detector name: '%s' for %d donuts (defocal type: '%s')",
+            detectorName,
+            numDonuts,
+            defocalType,
+        )
+
+        # Add missing telescope coordinate columns required by downstream
+        # aggregation
+        # These are not generated by neural network, so use default values
+        donutData["thx_ccs"] = [np.nan] * numDonuts  # Telescope X in Camera Coordinate System
+        donutData["thy_ccs"] = [np.nan] * numDonuts  # Telescope Y in Camera Coordinate System
+        donutData["thx_ocs"] = [np.nan] * numDonuts  # Telescope X in Observatory Coordinate System
+        donutData["thy_ocs"] = [np.nan] * numDonuts  # Telescope Y in Observatory Coordinate System
+        donutData["th_n"] = [np.nan] * numDonuts  # Telescope North coordinate
+        donutData["th_w"] = [np.nan] * numDonuts  # Telescope West coordinate
+
+        # Create the table
+        donutTable = QTable(donutData)
+
+        # Add units matching standard format
+        donutTable["coord_ra"].unit = u.rad  # Use radians like standard
+        donutTable["coord_dec"].unit = u.rad  # Use radians like standard
+        # Note: centroidX and centroidY are kept as plain floats to avoid
+        # Point2D constructor issues
+        donutTable["fx"].unit = u.deg
+        donutTable["fy"].unit = u.deg
+
+        # Add comprehensive visitInfo metadata using utility function
+        try:
+            donutTable, visitInfoMetadata = addVisitInfoToCatTable(exposure, donutTable)
+            # Now we have access to the comprehensive metadata dictionary
+            # This replaces the need for _getExposureMetadata in some cases
+        except AttributeError as e:
+            # Missing visitInfo or its attributes
+            self.log.warning(
+                "Could not extract comprehensive visitInfo metadata: %s. Using minimal visitInfo structure.",
+                e,
+            )
+            # Provide minimal visitInfo structure
+            try:
+                visitId = int(exposure.visitInfo.id)
+            except (AttributeError, ValueError, TypeError):
+                self.log.warning(
+                    "Could not extract visit ID from exposure.visitInfo, using fallback -1"
+                )
+                visitId = self.UNKNOWN_INT
+
+            donutTable.meta["visit_info"] = self._getDefaultVisitInfoDict(visitId)
+        except RuntimeError as e:
```

Comment:
Here you are repeating exactly the same thing as in the except above, you should consolidate this